hashicorp/dev-portal
# feat: bookmarks sidebar layout

## 🔗 Relevant links



- [Preview link](https://dev-portal-git-ksbookmarks-sidebar-layout-hashicorp.vercel.app) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1202832753754544) 🎟️
- [Design File](https://www.figma.com/file/VD7ahvXuXWJApeGnhbW4hv/Developer?node-id=12138%3A127457)

## 🗒️ What

This PR sets up the bookmarks view to use the sidebar / sidecar layout & creates a custom sidebar to handle the unique needs of this view. 

## 🛠️ How

~~I had to add a prop to the sidebar sidecar layout so the mobile authenticated disclosure dropdown could be opened on initial state. This is prop-drilly, but works.~~ Adjusting things to the mobile menu authentication component forces the dropdown disclosure to be open on all 'profile' routes. 

I chose to create a custom sidebar for this view since it does not behave the same as the other docs/tutorial sidebars. The designs don't call for walking nav levels and are specific about two different sidebar states between mobile and desktop views. This is because the mobile disclosure renders the sidebar items shown on desktop. 

## 📸 Design Screenshots

<img width="222" alt="Screen Shot 2022-08-24 at 5 04 03 PM" src="https://user-images.githubusercontent.com/36613477/186544971-5426ecce-e1e0-48d8-b7a9-0037741398ce.png">
<img width="731" alt="Screen Shot 2022-08-24 at 5 03 58 PM" src="https://user-images.githubusercontent.com/36613477/186544979-a0a2c79d-9d48-46d9-bc56-ccdde377cdbf.png">


## 🧪 Testing

- pull down this branch
- start the dev server and sign in 
- navigate to /profile/bookmarks
- you should see the proper sidebar layout
- with the correct nav items on deskop

- shrink down to mobile view 
- check the nav items and behavior matches the design spec. 

## Anything Else?
The `Resources` section on the sidebar is generated even though the designs don't render those. I think for consistency this is okay but probably something we should chat about in the long term.
